csharp/src/Ice/ConnectionFactory.cs

```diff
@@ -265,6 +299,15 @@ internal void RemoveAdapter(ObjectAdapter adapter)
             }
         }
 
+        internal void SetHintFailure(IConnector connector)
```

Comment:
Connections with a hint failure are still usable, just move down in the preference list, setting the hint failure prevents the following connections from going throw the same path, we can keep hint failures for connection establish, I think it makes sense that if you get ONE with OtherReplica set, the next attempt to use this object have into account the previous failure and prefers a replica that doesn't have a hint failure.

Reply:
I'm still not quite convinced we should do this. To me a remote retryable failure shouldn't affect the endpoint selection of other invocations since the retryable failure might really have nothing to do with the server's reachability (or at least we don't know this without further input from the application).

In other words, if I have a the following: 
```
var prx1 = FooPrx.Parse("ice+tcp://foo.com/myobject?alt-endpoint=foo2.com").Clone(connectionCached: false, endpointSelection: endpointSelectionType.Ordered);
var prx2 = FooPrx.Parse("ice+tcp://foo.com/myobject?alt-endpoint=foo2.com").Clone(connectionCached: false, endpointSelection: endpointSelectionType.Ordered);
try
{
     prx1.Invoke();
}
catch
{
}
prx2.Invoke();
```

I would expect the second invocation to still be sent to `foo.com` even if the first invocation triggered a retryable failure on `foo.com`. This is also what would occur if we didn't disable connection re-use (with `connectionCached : false`).

Reply:
I already update the code to do that, I think.

Reply:
One thing to keep in mind is that we change the cached connection, when we receive this exception, I assume this is fine.

Reply:
We should remove this method since we now only add entries from the outgoing connection factory.

Reply:
Actually, I think it might be good to still provide transport failure hints when an invocation fails because of a transport failure. I wouldn't add a hint failure for all kind of exceptions but just for exceptions that indicate reachability issues with the server (so only exceptions derived from TransportException).